### PR TITLE
Feature Invoke Attrs and multi-sign

### PR DESCRIPTION
**Pull request by localhuman**
**What current issue(s) does this address, or what feature is it adding?**
- Adds ability to attach tx attributes to SC invokes
- Adds the abililty to attach an arbitrary number of signing 'owners' to a transaction
- Other various fixes

**How did you solve this problem?**
- Add optional `invoke_attrs` and `owners` parameters to the invoke routines

**How did you make sure your solution works?**
- tests

**Are there any special changes in the code that we should be aware of?**
- no

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [ ] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

File: neo/Core/Block.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def Equals(self, other):
         return self.Hash == other.Hash
 
     @staticmethod
-    def FromTrimmedData(byts, index=None, transaction_method=None):
+    def FromTrimmedData(byts):
```
Comment (ixje) [inline neo/Core/Block.py]: please also remove the parameters from the docsstrings

File: neo/Implementations/Wallets/peewee/UserWallet.py [hunk 2]
```diff
@@ -401,6 +398,7 @@ def OnSaveTransaction(self, tx, added, changed, deleted):
 
     def OnCoinsChanged(self, added, changed, deleted):
 
+        some = False
```
Comment (ixje) [inline neo/Implementations/Wallets/peewee/UserWallet.py]: delete left over/unused var

File: neo/Core/FunctionCode.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def __init__(self, script=None, param_list=None, return_type=None, contract_prop
 
         if return_type and not isinstance(return_type, int) and len(return_type) > 1:
             return_type = return_type[0:1]
-        elif isinstance(return_type, int) and not 0 < return_type <= 255:
+        elif isinstance(return_type, int) and return_type > 255:
```
Comment (ixje) [inline neo/Core/FunctionCode.py]: just something I observed; `FunctionCode` is often initialized as `FunctionCode()` throughout the code base. This creates a return_type of `None`. There is no check that makes sure it becomes a valid return type. Perhaps we should make the default value to be `0xff` (void)?

File: neo/Wallets/NEP5Token.py [hunk 4]
```diff
@@ -244,7 +244,7 @@ def Approve(self, wallet, owner_addr, requestor_addr, amount):
 
         return tx, fee, results
 
-    def Mint(self, wallet, mint_to_addr, attachment_args):
+    def Mint(self, wallet, mint_to_addr, attachment_args, invoke_attrs=None):
```
Comment (ixje) [inline neo/Wallets/NEP5Token.py]: please update the docstring for the new argument

**Comment by localhuman** [comment]
Thanks for the feedback!  Will do that and also add some tests

**Pull request merged**